### DISCUSSION TITLE
menu with particles.js

### DISCUSSION THREAD

**Discussion by nenadjozic** [matteobruni/tsparticles#4007] [Q&A] [answered]
Is it possible to create menu with particles.js with clickable images moving that opens popup on click? any documentation?

**Comment by matteobruni** [comment, marked as answer]
Hi @nenadjozic,

I've created a sample with the click handler that adds some data to particles: <https://codepen.io/matteobruni/pen/yLvyKJd>

You have to add particles manually, because particles, for now, can't store any additional data from the configuration.

The `addParticle` method used in the sample has 3 parameters: exact position, custom options (the same values that `particles` section accepts), group to load.

You can set undefined for position, so a random one will be used, the options can be used to change the shape, like in the sample, and the group can be ignored.

**Reply by nenadjozic** [reply]
I try your sample, but sometimes randomly manually added particles disappear after one second or not appear at all. Whats wrong?

**Reply by matteobruni** [reply]
I don’t know, do you have a reproducible sample?

**Reply by nenadjozic** [reply]
i dont have online sample, but when i try to copy your sample from https://codepen.io/matteobruni/pen/yLvyKJd to my local server, still not working properly...

**Reply by matteobruni** [reply]
It's better if you create a sample that I can see so I can find any error

**Reply by nenadjozic** [reply]
here is the sample
https://ssac.org.rs/test2/
try to refres few times

**Reply by matteobruni** [reply]
I've seen that on your site you have both tsParticles and particles.js, I think that could create some conflict. Anyway, I've created a similar sample on CodePen here: https://codepen.io/matteobruni/pen/PoQxpPK that doesn't have particles.js loaded and it seems to be working, can you test it with your browser to see if it works for you as well?

**Comment by nenadjozic** [comment]
thx, how to add text label below or on top of each particle (name or something)? 
I dont need solution, i prefer hint or links

**Reply by matteobruni** [reply]
You need a custom shape for that, these are the existing shapes: https://github.com/matteobruni/tsparticles/tree/main/shapes

A custom shape sample:

https://codepen.io/matteobruni/pen/vYOMmGr
